core/trino-main/src/test/java/io/trino/operator/output/TestPagePartitioner.java

```diff
@@ -150,120 +148,134 @@ public void testOutputForSimplePage()
     @Test
     public void testOutputForEmptyPage()
     {
-        PartitionedOutputOperator partitionedOutputOperator = partitionedOutputOperator(BIGINT).build();
+        PartitionedOutputOperator partitionedOutputOperator = pagePartitioner(BIGINT).buildPartitionedOutputOperator();
         Page page = new Page(createLongsBlock(ImmutableList.of()));
 
-        processPages(partitionedOutputOperator, page);
+        partitionedOutputOperator.addInput(page);
+        partitionedOutputOperator.finish();
 
         List<Object> partitioned = readLongs(outputBuffer.getEnqueuedDeserialized(), 0);
         assertThat(partitioned).isEmpty();
     }
 
-    @Test
-    public void testOutputForPageWithNoBlockPartitionFunction()
+    @Test(dataProvider = "partitioningMode")
+    public void testOutputForSimplePage(PartitioningMode partitioningMode)
```

Comment:
What is the purpose of this test? Why all rows landed in partition number 0?

Reply:
this tests that the output of the `PagePartitioner` contains exactly the input when all partitions are concatenated (so regardless of the partitioning).
Not all rows go to partition 0. See comment below.

Reply:
> this tests that the output of the PagePartitioner contains exactly the input when all partitions are concatenated (so regardless of the partitioning).

`testProducesSameNumberOfRows`?

Reply:
@sopel39 this tests the actual values in the output page matches input page/block so maybe `testOutputEqualsInput`?